marimo/_utils/parse_dataclass.py

```diff
@@ -87,12 +90,24 @@ def _build_value(value: Any, cls: Type[T]) -> T:
         return value  # type: ignore[no-any-return]
 
 
-def build_dataclass(value: dict[Any, Any], cls: Type[T]) -> T:
+def build_dataclass(values: dict[Any, Any], cls: Type[T]) -> T:
+    """Returns instance of dataclass [cls] instantiated from [values]."""
+
+    if not isinstance(values, dict):
+        raise ValueError(
+            "value passed to build_dataclass needs to be a dictionary"
+        )
+
     types = get_type_hints(cls)
 
+    snake_cased_values = {to_snake(k): v for k, v in values.items()}
+    if not snake_cased_values.keys() <= types.keys():
```

Comment:
this is probably a good thing for us to have. any chance you can add a test case for a subset as well as a superset (and the superset should pass)

Reply:
Implicitly, it is already tested in `test_parse_dataclass.py`:
```
        nested = Nested(config=ConfigTwo(gpu=True))
        parsed = parse_raw(serialize(nested), Nested)
```
will fail here when `cls` is `ConfigOne` (first part of `Union[ConfigOne, ConfigTwo]` that it is tested against) because `ConfigOne` expects `types.keys() = dict_keys(['disabled'])` but `nested.config`, i.e. `ConfigTwo`, only provides `gpu` as key in `snake_cased_values`. 

And in `test_build_optional()` it passes (with `types.keys()` being `dict_keys(['x'])`, `snake_cased_values.keys()` being empty).

(We could add more explicit tests, of course.)

Reply:
ah great, ok thanks - yea we can add explicit tests but maybe in a followup